### PR TITLE
DEPR: inplace setting of Categorical.categories

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

I thought we had gotten all of these, but apparently not.

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -746,6 +746,15 @@ def categories(self):
 
     @categories.setter
     def categories(self, categories):
```
Comment (jreback) [inline pandas/core/arrays/categorical.py:748]: cna you instead make a _categories and use that internally?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/categorical.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:748]: we _really_ shouldn't be modifying dtypes inplace at all.  absolute footgun.

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -1062,11 +1071,19 @@ def rename_categories(self, new_categories, inplace=no_default):
         cat = self if inplace else self.copy()
 
         if is_dict_like(new_categories):
-            cat.categories = [new_categories.get(item, item) for item in cat.categories]
+            with catch_warnings():
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:1074]: once the deprecation in this method is enforced, it won't be using it at all.  introducing a new private property in the interim just makes things more complicated.

**Comment by jbrockmendel** [comment]
hmm the suggested alternative is to use rename_categories, but that will make a copy, whereas what we really want to offer is to make a view.

**Comment by simonjayhawkins** [comment]
moving to 1.5

**Comment by jbrockmendel** [comment]
Mothballing while troubleshooting locally.

**Pull request closed without merging**